src/lib/hooks/useValidationModel.ts

```diff
@@ -61,5 +62,5 @@ export function useValidationModel(): ValidationModel {
         },
         isValidationInProgress: () => currentlyValidatingNamespaces.length > 0,
         setNamespacesCurrentlyValidating
-    }
+    }), [currentlyValidatingNamespaces.length, validationModel]);
```

Comment:
This is the current problem:
- `[]` will cause the Leafs to render independently - great! But causes bugs:
  - validation errors and progress doesn't update.
- `[currentlyValidatingNamespaces.length, validationModel]` will cause Leafs to over-render.

Reply:
An approach that might work is to extract all getter functions from validationModel. If the getter functions are module scope and you pass the validation model to them then in theory validation model can be used similar to model - the design of model and usage of model is memo-able.

Reply:
After extracting the getter functions:
`[currentlyValidatingNamespaces.length, validationModel]` -> `[currentlyValidatingNamespaces, validationModel]`
(it's no better)